code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -766,9 +770,9 @@ func (sc *StorageSmartContract) challengeFailed(
 	}
 
 	logging.Logger.Info("Challenge failed", zap.String("challenge", cab.challenge.ID))
-
+	validators := getRandomSubSlice(cab.validators, validatorsRewarded, balances.GetBlock().GetRoundRandomSeed())
 	err := sc.blobberPenalty(
-		cab.alloc, cab.latestCompletedChallTime, cab.blobAlloc, cab.validators,
+		cab.alloc, cab.latestCompletedChallTime, cab.blobAlloc, validators,
```

Comment:
why do we penalize only subset of validators?

Reply:
Validators do not get penalised; blobbers do. The cab.blobAlloc parameter contains the blobbers to be slashed.

The validator parameter gives the validators that are going to receive a reward for scoring this particular challenge.